docs/runtimes/image-build.md

```diff
@@ -0,0 +1,192 @@
+
+In this case, the entire code repository will be retrieved from git as part of the image-building process, and cloned
+into the built image. This is recommended when the function relies on code spread across multiple files and also is 
+usually preferred for production code, since it means that the code of the function is stable, and further modifications 
+to the code will not cause instability in deployed images.
+
+During the development phase it may be desired to retrieve the code in runtime, rather than re-build the function
+image every time the code changes. To enable this, use {py:func}`~mlrun.projects.MlrunProject.set_source()` which
+gets a path to the source (can be a git repository or a tar or zip file) and set `pull_at_runtime=True`.
+
+### Using a private Docker registry
+By default, images will be pushed to the registry configured during MLRun deployment, using the registry credentials 
+configured. 
+
+To push resulting images to a different registry, the registry URL should be provided in the `image` parameter. If
+the registry requires credentials, a k8s secret containing these credentials should be created, and its name should be
+passed in the `secret_name` parameter.
+
+When using ECR as registry, MLRun uses Kaniko's ECR credentials helper, in which case the secret provided should contain
+AWS credentials needed to create ECR repositories, as described [here](https://github.com/GoogleContainerTools/kaniko#pushing-to-amazon-ecr).
+MLRun detects automatically that the registry is an ECR registry (based on its path) and configures Kaniko to
+use the ECR helper. For example:
+
+```python
+# AWS credentials stored in a k8s secret -
+# kubectl create secret generic ecr-credentials --from-file=<path to .aws/credentials>
+
+project.build_function(
+    "trainer",
+    image="<aws_account_id>.dkr.ecr.us-east-2.amazonaws.com/myrepo/image:v1",
+    secret_name="ecr-credentials",
+)
+```
+
+### Build environment variables
+It is possible to pass environment variables that will be set in the Kaniko pod that executes the build. This 
+may be useful to pass important information needed for the build process. The variables are passed as a dictionary in
+the `builder_env` parameter, for example:
+
+```python
+project.build_function(
+   ...
+   builder_env={"GIT_TOKEN": token},
+)
+```
+
+## Deploying nuclio functions
+When using nuclio functions, the image build process is done by nuclio as part of the deployment of the function. 
+Most of the configurations mentioned in this page are available for nuclio functions as well. To deploy a nuclio 
+function, use {py:func}`~mlrun.projects.deploy_function()` instead of using 
+{py:func}`~mlrun.projects.build_function()` and {py:func}`~mlrun.projects.run_function()`.
+
+## Creating default Spark runtime images
+When using Spark to execute code, either using a Spark service (remote-spark) or the Spark operator, an image is 
```

Comment:
When using Spark to execute code, either using a Spark service (remote-spark) or the Spark operator, an image is 
required that contains both Spark binaries and dependencies, and MLRun code and dependencies.